crates/eth-rpc/src/rpc.rs

```diff
@@ -0,0 +1,42 @@
+use jsonrpsee::RpcModule;
+use kakarot_rpc_core::client::config::{JsonRpcClientBuilder, StarknetConfig};
+use kakarot_rpc_core::client::KakarotClient;
+use starknet::providers::jsonrpc::HttpTransport;
+use starknet::providers::JsonRpcClient;
+
+use crate::api::eth_api::EthApiServer;
+use crate::api::kakarot_custom_api::KakarotCustomApiServer;
+use crate::servers::eth_rpc;
+use crate::servers::eth_rpc::KakarotEthRpc;
+use crate::servers::kakarot_custom_rpc::{self, KakarotCustomRpc};
+
+pub struct KakarotRpc {
+    eth_rpc_module: RpcModule<KakarotEthRpc<JsonRpcClient<HttpTransport>>>,
+    kakarot_custom_rpc_module: RpcModule<KakarotCustomRpc<JsonRpcClient<HttpTransport>>>,
+}
+
+impl KakarotRpc {
+    pub fn new(starknet_config: StarknetConfig) -> Self {
+        let eth_rpc_module =
+            eth_rpc::KakarotEthRpc::new(Box::new(KakarotRpc::get_kakarot_client(starknet_config.clone()))).into_rpc();
+        let kakarot_custom_rpc_module =
+            kakarot_custom_rpc::KakarotCustomRpc::new(Box::new(KakarotRpc::get_kakarot_client(starknet_config)))
+                .into_rpc();
+
+        Self { eth_rpc_module, kakarot_custom_rpc_module }
+    }
+
+    pub fn get_rpc_module(&self) -> RpcModule<()> {
```

Comment:
can we add rustdoc comments
akin to what @jobez did in his recent PRs:)?

just explain the general flow of things

Reply:
gpt does a decent job at this, but give yourself time to read through it yourself before you push